pkg/collector/corechecks/containers/containerd/containerd.go

```diff
@@ -214,25 +226,11 @@ func computeMetrics(sender aggregator.Sender, cu cutil.ContainerdItf, fil *ddCon
 
 		currentTime := time.Now()
 		computeUptime(sender, info, currentTime, tags)
-		computeMem(sender, metrics.Memory, tags)
-
-		ociSpec, err := cu.Spec(ctn)
-		if err != nil {
-			log.Warnf("Could not retrieve OCI Spec from: %s: %v", ctn.ID(), err)
-		}
-
-		var cpuLimits *specs.LinuxCPU
-		if ociSpec != nil && ociSpec.Linux != nil && ociSpec.Linux.Resources != nil {
-			cpuLimits = ociSpec.Linux.Resources.CPU
-		}
-		computeCPU(sender, metrics.CPU, cpuLimits, info.CreatedAt, currentTime, tags)
-
-		if metrics.Blkio.Size() > 0 {
-			computeBlkio(sender, metrics.Blkio, tags)
-		}
 
-		if len(metrics.Hugetlb) > 0 {
-			computeHugetlb(sender, metrics.Hugetlb, tags)
+		if linuxMetrics != nil {
+			computeLinuxSpecificMetrics(linuxMetrics, sender, cu, ctn, info.CreatedAt, currentTime, tags)
+		} else if windowsMetrics := windowsStats.GetWindows(); windowsMetrics != nil {
+			computeWindowsSpecificMetrics(windowsMetrics, sender, tags)
```

Comment:
I think this can be combined with the switch block above.

Reply:
Good point. That allows us to remove some conditionals. I had to move the tags part up, but I don't think that matters.
I added this in a separate commit :+1: